packages/fixie/src/agent.ts

```diff
@@ -317,30 +345,213 @@ export class FixieAgent {
+      // to the provided local port via localhost.run. The subprocess returns a
+      // stream of JSON responses, one per line, with the external URL of the tunnel
+      // as it changes.
+      const subProcess = spawn(
+        'ssh',
+        [
+          '-R',
+          // N.B. 127.0.0.1 must be used on Windows (not localhost or 0.0.0.0)
+          `80:127.0.0.1:${port}`,
+          '-o',
+          // Need to send keepalives to prevent the connection from getting chopped
+          // (see https://localhost.run/docs/faq#my-connection-is-unstable-tunnels-go-down-often)
+          'ServerAliveInterval=59',
+          '-o',
+          'StrictHostKeyChecking=accept-new',
+          'nokey@localhost.run',
+          '--',
+          '--output=json',
+        ],
+        { signal }
+      );
+      subProcess.stdout.setEncoding('utf8');
+
+      // Every time the subprocess emits a new line, we parse it as JSON ans
+      // extract the 'address' field.
+      let currentLine = '';
+      subProcess.stdout.on('data', (chunk: any) => {
+        // We need to do buffering since the data we get from stdout
+        // will not necessarily be line-buffered. We can get 0, 1, or more complete
+        // lines in a single chunk.
+        currentLine += chunk;
+        let newlineIndex;
+        while ((newlineIndex = currentLine.indexOf('\n')) !== -1) {
+          const line = currentLine.slice(0, newlineIndex);
+          currentLine = currentLine.slice(newlineIndex + 1);
+          // Parse sdata as JSON.
+          const pdata = JSON.parse(line);
+          // If pdata has the 'address' field, yield it.
+          if (pdata.address) {
+            // We can't yield until the parent has set up the yieldValue callback.
+            if (yieldValue) {
+              yieldValue(`https://${pdata.address}`);
             }
           }
         }
-      `,
-      variables: { agentId: this.agentId },
-    });
+      });
+      if (debug) {
+        subProcess.stderr.on('data', (sdata: string) => {
+          console.error(`🚇 Tunnel stderr: ${sdata}`);
+        });
+        subProcess.on('close', (returnCode: number) => {
+          console.log(`🚇 Tunnel child process exited with code ${returnCode}`);
+        });
+      }
+      // Now we wait for the subprocess to yield a new address, which we then
+      // re-yield to the caller of this function.
+      while (true) {
+        const tunnelAddress = await whenYielded;
+        if (tunnelAddress === '') {
+          break;
+        }
+        yield tunnelAddress;
+        yieldValue = null;
+        whenYielded = new Promise<string>((resolve) => {
+          yieldValue = resolve;
+        });
+      }
+    }
 
-    return result.data.agentById.currentRevision as AgentRevision;
+    // This is an iterator which yields the public URL of the tunnel where the agent
+    // can be reached by the Fixie service. The tunnel address can change over time.
+    let deployment_urls_iter;
```

Comment:
```suggestion
    let deploymentUrlsIter;
```

this ain't python y'all